c/src/core/util.c

```diff
@@ -38,7 +38,7 @@ ssize_t pn_quote_data(char *dst, size_t capacity, const char *src, size_t size)
   for (unsigned i = 0; i < size; i++)
   {
     uint8_t c = src[i];
-    if (isprint(c)) {
+    if (c < 128 && isprint(c)) {
```

Comment:
The bit I expect is directly comparable from proton-j uses:
`if (c > 31 && c < 127 && c != '\\')`

The < 127 and < 128 difference isnt important since DEL (127) isnt printable, but I'd note that proton-j also excludes \ (92) from direct inclusion since it could clash with the \x prefix used for the alternative hex-format output case.

Reply:
Yes, it makes lots of sense to make this reversible!